--- linux-6.6.46/include/linux/w1.h
+++ linux-6.6.46/include/linux/w1.h
@@ -145,6 +145,9 @@
 
 	void		(*write_block)(void *, const u8 *, int);
 
+	void		(*touch_block)(void *, u8 *, int);
+#define HAVE_W1_TOUCH_BLOCK
+
 	u8		(*triplet)(void *, u8);
 
 	u8		(*reset_bus)(void *);
--- linux-6.6.46/drivers/w1/w1_io.c
+++ linux-6.6.46/drivers/w1/w1_io.c
@@ -273,6 +273,12 @@
 	int i, j;
 	u8 tmp;
 
+#ifdef HAVE_W1_TOUCH_BLOCK
+	if (dev->bus_master->touch_block) {
+		dev->bus_master->touch_block(dev->bus_master->data, buf, len);
+		return;
+	}
+#endif
 	for (i = 0; i < len; ++i) {
 		tmp = 0;
 		for (j = 0; j < 8; ++j) {
